scripts/promise_batch_imports.py

```diff
@@ -57,7 +75,9 @@ def map_book_to_olbook(book, promise_id):
         'publishers': [book['ProductJSON'].get('Publisher') or '????'],
         'source_records': [f"promise:{promise_id}:{sku}"],
         # format_date adds hyphens between YYYY-MM-DD
-        'publish_date': publish_date and format_date(publish_date) or '????',
+        'publish_date': publish_date
+        and format_date(publish_date, USE_ONLY_YEAR)
```

Comment:
- it's weird for a function called `format_date` to take in a regex to test the string its given to determine if it should only output the year :P It's doing more than formatting in that case!
- Right now the check is simple so we don't need it to be as configurable. If they change we'll adapt and make the code more generic :)

```suggestion
        and format_date(publish_date, only_year=publish_date[-4:] in ('0000', '0101'))
```

Reply:
Also instead of and `X and Y or Z` switch to `Y if X else Z` that's a little more pythonic :)